### ISSUE TITLE
 obj_mask[b] = ~ pred_best_iou;obj_mask[b, a, j, i] = 1怎么理解？

### ISSUE THREAD

**Issue by ccrop** [Tianxiaomo/pytorch-YOLOv4#301]
# set mask to zero (ignore) if pred matches truth
 obj_mask[b] = ~ pred_best_iou

 for ti in range(best_n.shape[0]): #多少个gt label
                if best_n_mask[ti] == 1:  #当前gtlabel 是否在当前层中，是则进行计算
                    i, j = truth_i[ti], truth_j[ti]  #当前gt label 的 x，y格坐标
                    a = best_n[ti]   #得到anchor 层

                    obj_mask[b, a, j, i] = 1  #  怎样理解？应该是0？
                    tgt_mask[b, a, j, i, :] = 1

**Comment by xlwan1132** [comment]
output[..., 4] *= obj_mask
target[..., 4] *= obj_mask
loss_obj += F.binary_cross_entropy(input=output[..., 4], target=target[..., 4], reduction='sum')
你结合他置信度loss计算来看就明白了，只计算负例和iou最大的
